### PR TITLE
Recognize AArch64 in traceformat

### PR DESCRIPTION
**Pull request by knn-k**
This commit adds AArch64 to the architectures recognized by traceformat.

Fixes: #18536

### CODE REVIEW

**Comment by knn-k** [comment]
This commit depends on https://github.com/eclipse/omr/pull/7208

**Comment by knn-k** [comment]
I see the traceformat output as shown below with this fix:

```
Processor information:
        Arch family:         AArch64
        Processor Sub-type:  Armv8
        Num Processors:      4
        Big Endian:          false
        Word size:           64
        Using Trace Counter: J9 timer
```

**Comment by knn-k** [comment]
Jenkins test sanity alinux64,amac jdk17 depends eclipse/omr#7208

**Comment by knn-k** [comment]
@0xdaryl FYI.

**Comment by knn-k** [comment]
@keithc-ca Would you review this PR, please?  Not in a hurry.

File: jcl/src/openj9.traceformat/share/classes/com/ibm/jvm/format/ProcessorSection.java [hunk 1]
```diff
@@ -80,9 +80,9 @@ protected void summary(BufferedWriter out) throws IOException
          * These String arrays refer to enum values in entries of the
          * ProcessorInfo structure defined in xhpi.h .
          */
-        String[] Archs      = { "Unknown", "x86", "S390", "Power", "IA64", "S390X", "AMD64", "RISCV"};
+        String[] Archs      = { "Unknown", "x86", "S390", "Power", "IA64", "S390X", "AMD64", "RISCV", "AArch64" };
         String[] SubTypes   = { "i486", "i586", "Pentium II", "Pentium III",
-            "Merced","McKinley", "PowerRS", "PowerPC", "GigaProcessor", "ESA", "Pentium IV", "T-Rex", "Opteron", "RV64G"};
+            "Merced","McKinley", "PowerRS", "PowerPC", "GigaProcessor", "ESA", "Pentium IV", "T-Rex", "Opteron", "RV64G", "Armv8-A" };
```
Comment (keithc-ca) [inline jcl/src/openj9.traceformat/share/classes/com/ibm/jvm/format/ProcessorSection.java:85]: Please indent this like line 87, add the missing space after the first comma and fold it so it's no longer than line 83.

File: jcl/src/openj9.traceformat/share/classes/com/ibm/jvm/trace/format/api/ProcessorSection.java [hunk 2]
```diff
@@ -39,8 +39,8 @@ public class ProcessorSection {
 	protected static final int LONG = 8; // length of a long in bytes
 	protected static final String SUM_TAB = "        ";
 
-	private final static String[] Archs = { "Unknown", "x86", "S390", "Power", "IA64", "S390X", "AMD64", "RISCV"};
-	private final static String[] SubTypes = { "i486", "i586", "Pentium II", "Pentium III", "Merced", "McKinley", "PowerRS", "PowerPC", "GigaProcessor", "ESA", "Pentium IV", "T-Rex", "Opteron", "RV64G"};
+	private final static String[] Archs = { "Unknown", "x86", "S390", "Power", "IA64", "S390X", "AMD64", "RISCV", "AArch64"};
+	private final static String[] SubTypes = { "i486", "i586", "Pentium II", "Pentium III", "Merced", "McKinley", "PowerRS", "PowerPC", "GigaProcessor", "ESA", "Pentium IV", "T-Rex", "Opteron", "RV64G", "Armv8-A"};
```
Comment (keithc-ca) [inline jcl/src/openj9.traceformat/share/classes/com/ibm/jvm/trace/format/api/ProcessorSection.java:43]: Please insert a space before `}` on each line. Line 43 is rather long: I think it's time to fold it.

File: runtime/oti/j9port.h [hunk 3]
```diff
@@ -981,6 +983,7 @@ typedef struct J9CacheInfoQuery {
 #define J9PORT_ARCH_S390 OMRPORT_ARCH_S390
 #define J9PORT_ARCH_S390X OMRPORT_ARCH_S390X
 #define J9PORT_ARCH_RISCV OMRPORT_ARCH_RISCV
+#define J9PORT_ARCH_AARCH64 OMRPORT_ARCH_AARCH64
```
Comment (keithc-ca) [inline runtime/oti/j9port.h:986]: Rather than add a new J9 macro, I suggest all `J9PORT_ARCH_*` be removed and `trclog.c` be updated to use the OMR names,

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: jcl/src/openj9.traceformat/share/classes/com/ibm/jvm/format/ProcessorSection.java (hunk 1 above)
Comment (knn-k) [reply jcl/src/openj9.traceformat/share/classes/com/ibm/jvm/format/ProcessorSection.java]: Changed the indentation and line folding.

File: jcl/src/openj9.traceformat/share/classes/com/ibm/jvm/trace/format/api/ProcessorSection.java (hunk 2 above)
Comment (knn-k) [reply jcl/src/openj9.traceformat/share/classes/com/ibm/jvm/trace/format/api/ProcessorSection.java]: I added spaces and changed line folding.

**Comment by knn-k** [comment]
Jenkins compile all jdk17

File: runtime/rastrace/trclog.c [hunk 4]
```diff
@@ -2465,27 +2465,30 @@ static UtProcessorInfo * getProcessorInfo(void)
 	initHeader(&ret->procInfo.header, "PIN", sizeof(UtProcInfo));
 	osarch = j9sysinfo_get_CPU_architecture();
 	if (NULL != osarch) {
-		if ((strcmp(osarch, J9PORT_ARCH_PPC) == 0)
-		|| (strcmp(osarch, J9PORT_ARCH_PPC64) == 0)
-		|| (strcmp(osarch, J9PORT_ARCH_PPC64LE) == 0)
+		if ((strcmp(osarch, OMRPORT_ARCH_PPC) == 0)
+		|| (strcmp(osarch, OMRPORT_ARCH_PPC64) == 0)
+		|| (strcmp(osarch, OMRPORT_ARCH_PPC64LE) == 0)
```
Comment (keithc-ca) [inline runtime/rastrace/trclog.c:2470]: While you're changing these lines, please put the constants on the left (like line 2489):
```c
		if ((0 == strcmp(osarch, OMRPORT_ARCH_PPC)
		|| (0 == strcmp(osarch, OMRPORT_ARCH_PPC64)
		|| (0 == strcmp(osarch, OMRPORT_ARCH_PPC64LE)
```

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: runtime/rastrace/trclog.c (hunk 4 above)
Comment (knn-k) [reply runtime/rastrace/trclog.c]: Changed as requested.

**Comment by knn-k** [comment]
Jenkins compile all jdk17

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**